### PR TITLE
canmatrix recipe

### PR DESCRIPTION
**Pull request by danielhrisca**
add canmtrix recipe

### CODE REVIEW

File: recipes/canmatrix/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    - python
+    - xlwt
+    - xlrd
+    - xlsxwriter
+    - pyyaml
+    - lxml
+    - future
+
+test:
+  imports:
```
Comment (synapticarbors) [inline recipes/canmatrix/meta.yaml]: You don't have to test import all of the dependencies. You typically should just have to import the package, so:

```
test:
   imports:
      - canmatrix
```

File: recipes/canmatrix/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+source:
+  
+  fn: 8ab2bc648ebeef5f9c6b7368d260cd9c1b38a7fc.zip
+  url: https://github.com/ebroecker/canmatrix/archive/8ab2bc648ebeef5f9c6b7368d260cd9c1b38a7fc.zip
+  sha256: {{ sha256 }}
+
+build:
+  number: 0
+  script: python -m pip install --no-deps --ignore-installed .
```
Comment (synapticarbors) [inline recipes/canmatrix/meta.yaml]: It looks like there are CLI entry points:

https://github.com/ebroecker/canmatrix/blob/development/setup.py#L86-L87

Please include the appropriate entry for them:

https://conda.io/docs/user-guide/tasks/build-packages/define-metadata.html#python-entry-points

and add tests:

https://conda.io/docs/user-guide/tasks/build-packages/define-metadata.html#test-commands

File: recipes/canmatrix/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  
+  fn: 8ab2bc648ebeef5f9c6b7368d260cd9c1b38a7fc.zip
+  url: https://github.com/ebroecker/canmatrix/archive/8ab2bc648ebeef5f9c6b7368d260cd9c1b38a7fc.zip
+  sha256: {{ sha256 }}
+
+build:
```
Comment (synapticarbors) [inline recipes/canmatrix/meta.yaml]: Is this package a candidate for `noarch: python`?

http://conda-forge.readthedocs.io/en/latest/meta.html#building-noarch-packages

File: recipes/canmatrix/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+        .arxml (very basic implementation)
+        .yaml (dump of the python object)
+        .sym
+        .xml fibex
+
+  doc_url: 
+  dev_url: https://github.com/ebroecker/canmatrix
+
+extra:
+  recipe-maintainers:
```
Comment (synapticarbors) [inline recipes/canmatrix/meta.yaml]: @ebroecker and @altendky -- are you ok being added as a co-maintainer of the `canmatrix` recipe on conda-forge?

https://conda-forge.org/docs/recipe.html#maintainer-role

**Comment by danielhrisca** [comment]
@synapticarbors adding _noarch_ breaks appveyor. The other requested changes have been made

**Comment by synapticarbors** [comment]
@danielhrisca - I'm not sure what's going on in the appveyor error. 

@conda-forge/staged-recipes - any idea what might be going on here?

**Comment by dougalsutherland** [comment]
I'm not sure, but something similar happened with `ablog`: https://github.com/conda-forge/staged-recipes/pull/5271#issuecomment-372243151

File: recipes/canmatrix/meta.yaml (hunk 4 above)
Comment (ebroecker) [reply recipes/canmatrix/meta.yaml]: I'm not sure if I can help here anything. But if there are things/issues which should be solved in canmatrix - please let us know.

**Comment by danielhrisca** [comment]
@synapticarbors how do we proceed now?

**Comment by synapticarbors** [comment]
Merging based on @dougalsutherland's advice. Hopefully there won't be any issues in the feedstock with the windows build.

**Comment by danielhrisca** [comment]
@synapticarbors 
Now CircleCI fails on the feedstock due to incorrect sha256, however I've double checked and the recipe sha256 is correct for the downloaded archive (the recipe was done based on a certain commit from the development branch of canmatrix)

https://circleci.com/gh/conda-forge/canmatrix-feedstock/2

**Comment by synapticarbors** [comment]
When I do the following:

```
$ wget https://github.com/ebroecker/canmatrix/archive/8ab2bc648ebeef5f9c6b7368d260cd9c1b38a7fc.zip
$ shasum -a 256 8ab2bc648ebeef5f9c6b7368d260cd9c1b38a7fc.zip
```

I get:

```
7ca330e7d1728190bb25155abf729527ab6bcf3314503e36a4737f20719ea052  8ab2bc648ebeef5f9c6b7368d260cd9c1b38a7fc.zip
```

So I can reproduce the error locally on my laptop. 

Can you put in a PR in the feedstock to fix this and then you should be able to merge it.

**Comment by danielhrisca** [comment]
That's strange for me, but true.


image
image


I had a zip from a few days back and it has the older sha256. 

How is this possible? 

I'll make the PR like you have suggested. Thank you!

**Comment by dougalsutherland** [comment]
Github commit archives I think can change checksums when github updates their version of `git` or something. This is part of why it's nicer to use official releases when they exist.

**Pull request merged**